docs/docs/dev_docs/contracts/compiling.md

```diff
@@ -1,22 +1,136 @@
 # Compiling contracts
 
-Please use the [TUTORIAL-TEMPLATE](../../TUTORIAL_TEMPLATE.md) for standalone guides / tutorials.
+Once you have written your [contracts](../contracts/main.md) in Noir, you will need to compile them into an artifact in order to use them. In this guide we will cover how to do so, both using the CLI and programmatically, as well as generating helper typescript and Noir interfaces for interacting with them.
 
-:::danger TODO
-TODO: this entire page
+## Prerequisites
+
+You will need the Noir build tool `nargo`, which you can install via [`noirup`](https://github.com/noir-lang/noirup). Make sure you install the `aztec` version of nargo:
+
+```bash
+curl -L https://raw.githubusercontent.com/noir-lang/noirup/main/install | bash
+noirup -v aztec
+```
+
+:::info
+You can re-run `noirup -v aztec` whenever you want to update to the latest version of Noir supported by the Aztec Network.
 :::
-## Compiling a Noir Contract
 
-You can use the `master` branch of Noir/nargo (hooray).
+## Compile using the CLI
+
+To compile using the Aztec CLI, first install it:
+
+`npm install -g @aztec/cli`
+
+Then run the `compile` command with the path to your Noir contract project:
+
+```
+aztec-cli compile ./src/noir
```

Comment:
I just tried this command, myself, and I'm getting `error: unknown command 'compile'`. Is CI not pushing correctly to master? (I had similar problems when trying to follow the quick start guide that you wrote yesterday #1579 )

Reply:
Indeed, CLI is being released only on tagged releases